### ISSUE TITLE
Global search beta follow up

### ISSUE THREAD

**Issue by Fenn-CS** [nextcloud/server#41381]
Follow up issues from https://github.com/nextcloud/server/pull/40823

### TODO

**High priority:**
- [x] Typing in the user search filter doesn't show all the users, it seems that it shows only the preloaded ones
- [x] When filtering the local view (e.g. users) it's impossible to reset the view. When reopening the search modal, it should be possible to reset the search.
- [x] Search result links that should be improved to be actual links for a11y reasons

**Other:** 
- [x] Icon inversion which does not work well now since icons come in white or black instead of one consistent color. (Black icons turn white and white icons turn black, leading to to white on white background issues)
- [x] Use the new people filter (selector) that @marcoambrosini is working on (Current people filter not completed) - DONE, however, NcPopover which is used has issues with closing triggers.
- [x] Update back-end API (if ability not already present) to show "Search in App" dynamically.
- [x] Remove person filter
- [x] BUG: Clicking outside the search modal needs two clicks on the magnifier icon to get the modal back.
- [x] BUG: Some app icons are white on white.
- [x] BUG: Modified timestamp bubbles change size on hover, have no pointing cursor.
- [x] BUG: Generally no feedback for no results.
- [x] BUG: Search people has white text on white background.
- [x] BUG: Search people has a tiny user bubble, barely wider than the avatar.
- [x] BUG: Search people No result goes to the top, overlaps other input.
- [x] BUG: Search people has no effect at all. The search string doesn't contain a person identifier.
- [x] Avatars are square. I guess that's because all previews are square so this makes sense *technically*. For a user, this is inconsistent with *avatars*, though
    ![image](https://github.com/nextcloud/server/assets/1374172/5dfcef83-ae9b-47cf-b861-aa79f65c34e2)
- [x] When previews do not exist (endpoint returns 404) the browser renders a placeholder for the broken image
    ![Bildschirmfoto vom 2023-11-14 14-23-13](https://github.com/nextcloud/server/assets/1374172/c677f3c0-38a4-42e3-8faa-a632326d2e2d)
- [x] *x* on apps/person selection changes size on hover
- [x] No space for icons in modified (Fenn-CS : Best handled at component level, migrate to vue lib repo? "Do not reserve space for icon in NcActionButton if no icon is provided"???)
- [x] *x* on apps/person selection should have pointing cursor

![image](https://github.com/nextcloud/server/assets/1374172/eb7bd1f9-19f4-4e44-9258-b3f1e4d00ba3)
- [ ] People are not sorted, at least not by name
    ![image](https://github.com/nextcloud/server/assets/1374172/4a3d84d0-b708-49f9-b5d0-06728e3029c1)

- [ ] Accidentally clicked the magnifier on mobile and found out it's impossible to close the modal
![image](https://github.com/nextcloud/server/assets/1374172/2fb106aa-ad63-476b-b3c3-aa4931f08fb1)

Needs full page reload to get back to the content.

**Comment by ChristophWurst** [comment]
Noticed during testing today

- [ ] Avatars are square. I guess that's because all previews are square so this makes sense *technically*. For a user, this is inconsistent with *avatars*, though
    ![image](https://github.com/nextcloud/server/assets/1374172/5dfcef83-ae9b-47cf-b861-aa79f65c34e2)
- [ ] When previews do not exist (endpoint returns 404) the browser renders a placeholder for the broken image
    ![Bildschirmfoto vom 2023-11-14 14-23-13](https://github.com/nextcloud/server/assets/1374172/c677f3c0-38a4-42e3-8faa-a632326d2e2d)

**Comment by ChristophWurst** [comment]
- [ ] *x* on apps/person selection changes size on hover
- [ ] *x* on apps/person selection should have pointing cursor

![image](https://github.com/nextcloud/server/assets/1374172/eb7bd1f9-19f4-4e44-9258-b3f1e4d00ba3)

**Comment by ChristophWurst** [comment]
- [ ] People are not sorted, at least not by name
    ![image](https://github.com/nextcloud/server/assets/1374172/4a3d84d0-b708-49f9-b5d0-06728e3029c1)

**Comment by ChristophWurst** [comment]
- [ ] Search result have a href of `#`. Opening them in a new tab brings you to the same page you were when searching, not the search result. This is probably an a11y issue too.

**Comment by blizzz** [comment]
Should https://github.com/nextcloud/server/issues/41484 be linked here, too?

**Comment by Altahrim** [comment]
- [ ] Impossible to select a date in the future
- [ ] Search providers for admins like `users` should probably be hidden for non admins

**Comment by blizzz** [comment]
>     * [ ]  Search providers for admins like `users` should probably be hidden for non admins

Then there is the case with group admins who would have access to a portion of users  :smiley:

**Comment by ChristophWurst** [comment]
- [ ] Accidentally clicked the magnifier on mobile and found out it's impossible to close the modal
![image](https://github.com/nextcloud/server/assets/1374172/2fb106aa-ad63-476b-b3c3-aa4931f08fb1)

Needs full page reload to get back to the content.

**Comment by SystemKeeper** [comment]
- [ ] When clicking on a search result (e.g. Talk or Files) you don't jump to that file/conversation anymore, instead:
  ![image](https://github.com/nextcloud/server/assets/1580193/d617e83a-1c65-4e09-951b-496d2ffe4801)

**Comment by ChristophWurst** [comment]
> When clicking on a search result

```
TypeError: t.openResult is not a function
    click SearchResult.vue:1
    VueJS 4
    onClick NcListItem.mjs:171
    click NcListItem.mjs:223
    VueJS 3
```

Same for results of Mail and other search providers

**Comment by ChristophWurst** [comment]
- [ ] BUG modal can only be opened once, then it auto-closes

1) Click magnifier icon
2) Click outside the modal
3) Click magnifier icon again
4) *Nothing happens*
5) Click magnifier icon again
6) *Modal opens, closes again*
7) goto 3.

**Comment by Fenn-CS** [comment]
>  When clicking on a search result (e.g. Talk or Files) you don't jump to that file/conversation anymore, instead:

Resolved in : https://github.com/nextcloud/server/pull/41785

**Comment by ChristophWurst** [comment]
https://github.com/nextcloud/server/issues/41381#issuecomment-1813247040 still happens and it's quite frustrating to have to reload the page every time I mislick the notification icon.

**Comment by Altahrim** [comment]
- [ ] Person filter don't find all users.  It seems to load an initial list and only filter this list.

**Comment by jancborchardt** [comment]
- [ ] When using the "Filter current view" in account management, it is not possible to reset the search. E.g. on tech-preview, opening the modal back again makes it come up with an already cleared search input so it is not clearable. But the search term is still applied in the main content behind the modal, as well as for e.g. the Giphy provider inside the modal.

**Comment by jancborchardt** [comment]
- [ ] "Start typing in search" → "Start typing to search"
- [ ] When clearing the search term by marking the whole thing and deleting it all in one, the results stay below the emptycontent view:
    ![image](https://github.com/nextcloud/server/assets/925062/d8072e1f-3560-4fa9-a2a8-27d1cccd07c6)

**Comment by ChristophWurst** [comment]
> Accidentally clicked the magnifier on mobile and found out it's impossible to close the modal

![image](https://github.com/nextcloud/server/assets/1374172/125e77ec-a838-4494-b4ac-e0c0b514beaa)

**Comment by Fenn-CS** [comment]
@ChristophWurst we have this already https://github.com/nextcloud/server/issues/41422

**Comment by ChristophWurst** [comment]
Thanks. It was not clear from this ticket.
